### ISSUE TITLE
SEARCH LIKE returns suitable and unsuitable results

### ISSUE THREAD

**Issue by hosh1kage** [arangodb/arangodb#19630]
## My Environment

* __ArangoDB Version__: 3.9.11/3.10.9/3.11.2;
* __Deployment Mode__: Single Server;
* __Deployment Strategy__:  Manual Start in Docker;
* __Configuration__: Default, ARANGO_NO_AUTH = TRUE;
* __Infrastructure__: Own;
* __Operating System__: Ubuntu 22.04/Windows 11;
* __Total RAM in your machine__: 32Gb;
* __Disks in use__: SSD;
* __Used Package__: Docker - official Docker library;

## Component, Query & Data

__Affected feature__:

AQL query using web interface.

__AQL query__:

```
FOR letter IN SPLIT("абвгдеёжзиклмнопрстуфхцчшщъыьэюя", "")
    LET pattern = CONCAT('%а', letter, 'а%')
    LET filterResult = (FOR t IN test FILTER LIKE(t.name, pattern) RETURN t.name)
    LET searchResult = (FOR t IN testView SEARCH ANALYZER(LIKE(t.name, pattern), 'identity') RETURN t.name)
    RETURN {pattern, filterResult, searchResult}
```

__AQL explain and/or profile__:

Explain:

```
Query String (343 chars, cacheable: true):
 FOR letter IN SPLIT("абвгдеёжзиклмнопрстуфхцчшщъыьэюя", "")
     LET pattern = CONCAT('%а', letter, 'а%')
     LET filterResult = (FOR t IN test FILTER LIKE(t.name, pattern) RETURN t.name)
     LET searchResult = (FOR t IN testView SEARCH ANALYZER(LIKE(t.name, pattern), 'identity') RETURN
 t.name)
     RETURN {pattern, filterResult, searchResult}

Execution plan:
 Id   NodeType                  Est.   Comment
  1   SingletonNode                1   * ROOT
  2   CalculationNode              1     - LET #10 = [ "а", "б", "в", "г", "д", "е", "ё", "ж", "з", "и", "к", "л", "м", "н", "о", "п", "р", "с", "т", "у", ... ]   /* json expression */   /* const assignment */
  3   EnumerateListNode           32     - FOR letter IN #10   /* list iteration */
  4   CalculationNode             32       - LET pattern = CONCAT("%а", letter, "а%")   /* simple expression */
 21   SubqueryStartNode           32       - LET searchResult = ( /* subquery begin */
 13   EnumerateViewNode         1024         - FOR t IN testView SEARCH ANALYZER(LIKE(t.`name`, pattern), "identity")   /* view query */
 14   CalculationNode           1024           - LET #16 = t.`name`   /* attribute expression */
 22   SubqueryEndNode             32           - RETURN  #16 ) /* subquery end */
 19   SubqueryStartNode           32       - LET filterResult = ( /* subquery begin */
  6   EnumerateCollectionNode   1024         - FOR t IN test   /* full collection scan (projections: `name`)  */   FILTER LIKE(t.`name`, pattern)   /* early pruning */
  9   CalculationNode           1024           - LET #14 = t.`name`   /* attribute expression */   /* collections used: t : test */
 20   SubqueryEndNode             32           - RETURN  #14 ) /* subquery end */
 17   CalculationNode             32       - LET #18 = { "pattern" : pattern, "filterResult" : filterResult, "searchResult" : searchResult }   /* simple expression */
 18   ReturnNode                  32       - RETURN #18

Indexes used:
 none

Functions used:
 Name     Deterministic   Cacheable   Uses V8
 CONCAT   true            true        false

Optimization rules applied:
 Id   RuleName
  1   move-calculations-up
  2   move-filters-up
  3   move-calculations-up-2
  4   move-filters-up-2
  5   handle-arangosearch-views
  6   move-calculations-down
  7   move-filters-into-enumerate
  8   reduce-extraction-to-projection
  9   splice-subqueries

43 rule(s) executed, 1 plan(s) created, peak mem [b]: 0, exec time [s]: 0.00194
```

Profile:

```
Query String (343 chars, cacheable: false):
 FOR letter IN SPLIT("абвгдеёжзиклмнопрстуфхцчшщъыьэюя", "")
     LET pattern = CONCAT('%а', letter, 'а%')
     LET filterResult = (FOR t IN test FILTER LIKE(t.name, pattern) RETURN t.name)
     LET searchResult = (FOR t IN testView SEARCH ANALYZER(LIKE(t.name, pattern), 'identity') RETURN
 t.name)
     RETURN {pattern, filterResult, searchResult}

Execution plan:
 Id   NodeType                  Calls   Items   Filtered   Runtime [s]   Comment
  1   SingletonNode                 1       1          0       0.00001   * ROOT
  2   CalculationNode               1       1          0       0.00000     - LET #10 = [ "а", "б", "в", "г", "д", "е", "ё", "ж", "з", "и", "к", "л", "м", "н", "о", "п", "р", "с", "т", "у", ... ]   /* json expression */   /* const assignment */
  3   EnumerateListNode             1      32          0       0.00005     - FOR letter IN #10   /* list iteration */
  4   CalculationNode               1      32          0       0.00003       - LET pattern = CONCAT("%а", letter, "а%")   /* simple expression */
 21   SubqueryStartNode             1      64          0       0.00002       - LET searchResult = ( /* subquery begin */
 13   EnumerateViewNode             1      80          0       0.00113         - FOR t IN testView SEARCH ANALYZER(LIKE(t.`name`, pattern), "identity")   /* view query */
 14   CalculationNode               1      80          0       0.00004           - LET #16 = t.`name`   /* attribute expression */
 22   SubqueryEndNode               1      32          0       0.00001           - RETURN  #16 ) /* subquery end */
 19   SubqueryStartNode             1      64          0       0.00002       - LET filterResult = ( /* subquery begin */
  6   EnumerateCollectionNode       1      64        992       0.00092         - FOR t IN test   /* full collection scan (projections: `name`)  */   FILTER LIKE(t.`name`, pattern)   /* early pruning */
  9   CalculationNode               1      64          0       0.00001           - LET #14 = t.`name`   /* attribute expression */   /* collections used: t : test */
 20   SubqueryEndNode               1      32          0       0.00004           - RETURN  #14 ) /* subquery end */
 17   CalculationNode               1      32          0       0.00004       - LET #18 = { "pattern" : pattern, "filterResult" : filterResult, "searchResult" : searchResult }   /* simple expression */
 18   ReturnNode                    1      32          0       0.00004       - RETURN #18

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   move-calculations-up
  2   move-filters-up
  3   move-calculations-up-2
  4   move-filters-up-2
  5   handle-arangosearch-views
  6   move-calculations-down
  7   move-filters-into-enumerate
  8   reduce-extraction-to-projection
  9   splice-subqueries

Query Statistics:
 Writes Exec   Writes Ign   Scan Full   Scan Index   Cache Hits/Misses   Filtered   Peak Mem [b]   Exec Time [s]
           0            0        1024           48               0 / 0        992          98304         0.00335

Query Profile:
 Query Stage               Duration [s]
 initializing                   0.00000
 parsing                        0.00010
 optimizing ast                 0.00011
 loading collections            0.00001
 instantiating plan             0.00004
 optimizing plan                0.00033
 instantiating executors        0.00022
 executing                      0.00252
 finalizing                     0.00006
```

__Dataset__:

Test collection dataset:

```
[
  {
    "_key":"1043",
    "_id":"test/1043",
    "_rev":"_geon4XG---",
    "name":"ааа"
  },
  {
    "_key":"1047",
    "_id":"test/1047",
    "_rev":"_geooETG---",
    "name":"аба"
  },
  {
    "_key":"1051",
    "_id":"test/1051",
    "_rev":"_geoq40a---",
    "name":"ава"
  },
  {
    "_key":"1055",
    "_id":"test/1055",
    "_rev":"_georAnO---",
    "name":"ага"
  },
  {
    "_key":"1059",
    "_id":"test/1059",
    "_rev":"_georLli---",
    "name":"ада"
  },
  {
    "_key":"1063",
    "_id":"test/1063",
    "_rev":"_georUAy---",
    "name":"аеа"
  },
  {
    "_key":"1067",
    "_id":"test/1067",
    "_rev":"_geors_q---",
    "name":"аёа"
  },
  {
    "_key":"1071",
    "_id":"test/1071",
    "_rev":"_geory22---",
    "name":"ажа"
  },
  {
    "_key":"1075",
    "_id":"test/1075",
    "_rev":"_geor41a---",
    "name":"аза"
  },
  {
    "_key":"1079",
    "_id":"test/1079",
    "_rev":"_geos_Py---",
    "name":"аиа"
  },
  {
    "_key":"1083",
    "_id":"test/1083",
    "_rev":"_geosFEG---",
    "name":"ака"
  },
  {
    "_key":"1087",
    "_id":"test/1087",
    "_rev":"_geosLLC---",
    "name":"ала"
  },
  {
    "_key":"1091",
    "_id":"test/1091",
    "_rev":"_geosSFG---",
    "name":"ама"
  },
  {
    "_key":"1095",
    "_id":"test/1095",
    "_rev":"_geosX5a---",
    "name":"ана"
  },
  {
    "_key":"1099",
    "_id":"test/1099",
    "_rev":"_geosdye---",
    "name":"аоа"
  },
  {
    "_key":"1103",
    "_id":"test/1103",
    "_rev":"_geosi66---",
    "name":"апа"
  },
  {
    "_key":"1107",
    "_id":"test/1107",
    "_rev":"_geot-wC---",
    "name":"ара"
  },
  {
    "_key":"1111",
    "_id":"test/1111",
    "_rev":"_geotFKe---",
    "name":"аса"
  },
  {
    "_key":"1115",
    "_id":"test/1115",
    "_rev":"_geotLga---",
    "name":"ата"
  },
  {
    "_key":"1119",
    "_id":"test/1119",
    "_rev":"_geotS5a---",
    "name":"ауа"
  },
  {
    "_key":"1123",
    "_id":"test/1123",
    "_rev":"_geotYg6---",
    "name":"афа"
  },
  {
    "_key":"1127",
    "_id":"test/1127",
    "_rev":"_geoteUi---",
    "name":"аха"
  },
  {
    "_key":"1131",
    "_id":"test/1131",
    "_rev":"_geot3Oq---",
    "name":"аца"
  },
  {
    "_key":"1135",
    "_id":"test/1135",
    "_rev":"_geot9pO---",
    "name":"ача"
  },
  {
    "_key":"1139",
    "_id":"test/1139",
    "_rev":"_geouPTW---",
    "name":"аша"
  },
  {
    "_key":"1143",
    "_id":"test/1143",
    "_rev":"_geouccK---",
    "name":"аща"
  },
  {
    "_key":"1147",
    "_id":"test/1147",
    "_rev":"_geoun9G---",
    "name":"аъа"
  },
  {
    "_key":"1151",
    "_id":"test/1151",
    "_rev":"_geoutuK---",
    "name":"аыа"
  },
  {
    "_key":"1155",
    "_id":"test/1155",
    "_rev":"_geou0m6---",
    "name":"аьа"
  },
  {
    "_key":"1159",
    "_id":"test/1159",
    "_rev":"_geou6S2---",
    "name":"аэа"
  },
  {
    "_key":"1163",
    "_id":"test/1163",
    "_rev":"_geovB3O---",
    "name":"аюа"
  },
  {
    "_key":"1167",
    "_id":"test/1167",
    "_rev":"_geovHXy---",
    "name":"аяа"
  }
]
```

Test collection view links:

```
"links": {
  "test": {
    "analyzers": [
      "identity"
    ],
    "fields": {},
    "includeAllFields": true,
    "storeValues": "none",
    "trackListPositions": false
  }
}
```

__Size of your Dataset on disk__:

3Kb.

## Steps to reproduce

1. Create collection ```test```;
2. Fill collection ```test``` with attached dataset;
3. Create ```arangosearch``` view ```testView``` with attached links;
4. Execute attached AQL query;

__Problem__:

If you use ```FILTER LIKE``` and ```SEARCH LIKE``` to search substring in string with russian language content then only 1 suitable result is always returned, except 1 case: ```SEARCH LIKE``` with pattern ```%ара%``` returns 1 suitable and multiple unsuitable results. As you can see all results ```["аёа","ара","аса","ата","ауа","афа","аха","аца","ача","аша","аща","аъа","аыа","аьа","аэа","аюа","аяа"]``` have second letter whose code is not only equal to ```р``` but also greater than ```р```.

__Expected result__:
```SEARCH LIKE``` should return only suitable results.

**Comment by Simran-B** [comment]
I am able to reproduce the problem:

pattern | filterResult | searchResult
---|---|---
%ааа% | ааа | ааа
%аба% | аба | аба
%ава% | ава | ава
%ага% | ага | ага
%ада% | ада | ада
%аеа% | аеа | аеа
%аёа% | аёа | аёа
%ажа% | ажа | ажа
%аза% | аза | аза
%аиа% | аиа | аиа
%ака% | ака | ака
%ала% | ала | ала
%ама% | ама | ама
%ана% | ана | ана
%аоа% | аоа | аоа
%апа% | апа | апа
%ара% | ара | **аёа**, ара, **аса**, **ата**, **ауа**, **афа**, **аха**, **аца**, **ача**, **аша**, **аща**, **аъа**, **аыа**, **аьа**, **аэа**, **аюа**, **аяа**
%аса% | аса | аса
%ата% | ата | ата
%ауа% | ауа | ауа
%афа% | афа | афа
%аха% | аха | аха
%аца% | аца | аца
%ача% | ача | ача
%аша% | аша | аша
%аща% | аща | аща
%аъа% | аъа | аъа
%аыа% | аыа | аыа
%аьа% | аьа | аьа
%аэа% | аэа | аэа
%аюа% | аюа | аюа
%аяа% | аяа | аяа

Using a `collation` Analyzer with locale `ru` in the View and the query gives a different result, also with two peculiarities:

```js
FOR letter IN SPLIT("абвгдеёжзиклмнопрстуфхцчшщъыьэюя", "")
    LET pattern = CONCAT('%', TOKENS(CONCAT('а', letter, 'а'), "cyrl")[0], '%')
    LET filterResult = (FOR t IN test FILTER LIKE(TOKENS(t.name, "cyrl")[0], pattern) RETURN t.name)
    LET searchResult = (FOR t IN testView SEARCH ANALYZER(LIKE(t.name, pattern), 'cyrl') RETURN t.name)
    RETURN {patternHex: CONCAT_SEPARATOR(" ", REGEX_SPLIT(TO_HEX(pattern), "(..)")[* FILTER CURRENT]), filterResult, searchResult}
```

patternHex | filterResult | searchResult
---|---|---
25 27 06 06 06 01 07 07 25 | ааа | ааа
25 27 06 0c 06 01 07 07 25 | аба | аба
25 27 06 0e 06 01 07 07 25 | ава | ава
25 27 06 10 06 01 07 07 25 | ага | ага
25 27 06 1a 06 01 07 07 25 | ада | ада
25 27 06 28 06 01 07 07 25 | аеа | аеа
25 27 06 28 06 01 44 86 05 01 08 25 | аёа | аёа
25 27 06 2c 06 01 07 07 25 | ажа | ажа
25 27 06 32 06 01 07 07 25 | аза | аза
25 27 06 46 06 01 07 07 25 | аиа | аиа
25 27 06 54 06 01 07 07 25 | ака | ака
25 27 06 62 06 01 07 07 25 | ала | ала
25 27 06 72 06 01 07 07 25 | ама | ама
25 27 06 78 06 01 07 07 25 | ана | ана
25 27 06 d1 82 06 01 01 07 25 | аоа | аоа
25 27 06 d1 86 06 01 01 07 25 | апа | апа
25 27 06 d2 86 06 01 01 07 25 | ара | ара
25 27 06 d3 84 06 01 01 07 25 | аса | аса
25 27 06 d4 82 06 01 01 07 25 | ата | ата
25 27 06 d5 84 06 01 01 07 25 | ауа | ауа
25 27 06 d6 84 06 01 01 07 25 | афа | афа
25 27 06 d6 86 06 01 01 07 25 | аха | аха
25 27 06 d8 80 06 01 01 07 25 | аца | аца
25 27 06 d9 80 06 01 01 07 25 | ача | ача
25 27 06 db 82 06 01 01 07 25 | аша | аша
25 27 06 db 86 06 01 01 07 25 | аща | аща
25 27 06 dc 82 06 01 01 07 25 | аъа | аъа
25 27 06 dc 86 06 01 01 07 25 | аыа | аыа
25 27 06 dd 80 06 01 01 07 25 | аьа | аьа, _аэа_
25 27 06 dd 86 06 01 01 07 25 | аэа | аэа
25 27 06 de 80 06 01 01 07 25 | аюа | аюа, _аяа_
25 27 06 de 84 06 01 01 07 25 | аяа | аяа

**Comment by MBkkt** [comment]
@hosh1kage 
Thanks, https://github.com/iresearch-toolkit/iresearch/pull/552, fix will be available in 3.10.11, 3.11.4

**Comment by MBkkt** [comment]
Fixed for 3.9.12, 3.10.11, 3.11.4, devel
